dvc/repo/__init__.py

```diff
@@ -80,7 +80,7 @@ def __init__(self, root_dir=None):
 
         self.config = Config(self.dvc_dir)
 
-        self.scm = SCM(self.root_dir)
+        self.scm = SCM(root_dir, search_parent_directories=True)
```

Comment:
Hm, but what if I want to have no-scm repo? I've mentioned that if we get rid of subrepo config option, we might need it for no-scm repo, to clearly mark that we shouldn't try to look for scm's in parent dirs. Or I am missing something?

Reply:
It is a pretty synthetic use case, but still is possible. Just wondering if we should switch to a new behavior and if we need to support no-scm in git host repo.

Reply:
Ok, yes I did not noticed that.
So it seems that we have 2 options here:
1. Make config option for subrepo which will allow searching parent directories - no problems besides cluttering the config with additional option
2. Make no-scm config option for subrepo - that will break backwards compatibility with no-scm repos created before this change

Reply:
@pared It is fine to keep this conf option. When I commented on this I didn't consider `--no-scm` case. Also, as a 3rd option, you can prohibit using `--subrepo` in not Git case. It ups to you.

Reply:
`no-scm` will be stored as config option and read here, by default we will look for Git in parent dirs